include/IECore/Reader.h

```diff
@@ -37,6 +37,7 @@
 
 #include "IECore/CompoundObject.h"
 #include "IECore/Export.h"
+#include "IECore/FileNameParameter.h"
```

Comment:
I'm confused by this. There's a forward declaration lower down this file that should be enough for this header. Perhaps the fix belongs elsewhere?

Reply:
From what I remember the classes that subclass from FileNameParameter get tripped up in MSVC as it needs the full declaration, but this might've been from an old MSVC version or before the refactor, so I'll double-check and see if it still is a problem.

Reply:
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2664	'void IECore::intrusive_ptr_release(const IECore::RefCounted *)': cannot convert argument 1 from 'IECore::FileNameParameter *' to 'const IECore::RefCounted *' (compiling source file C:\Users\Alex\dev\cortex\src\IECore\CachedReader.cpp)	IECore	C:\Users\Alex\dev\gafferDependenciesInstall2017\include\boost-1_61\boost\smart_ptr\intrusive_ptr.hpp	97